angular/src/lib/data-table/data-table-scroll.component.ts

```diff
@@ -261,20 +261,24 @@ export class ScrollComponent implements AfterViewInit, OnDestroy, AfterViewCheck
 
   setScrollHeight() {
     if (this.scrollHeight && this.scrollBodyViewChild && this.scrollBodyViewChild.nativeElement) {
-      if (this.scrollHeight.indexOf('%') !== -1) {
+      if (this.scrollHeight.indexOf('%') !== -1) { // if scrollHeight is a % value
         let relativeHeight;
         this.scrollBodyViewChild.nativeElement.style.visibility = 'hidden';
         this.scrollBodyViewChild.nativeElement.style.height = '100px';
         const containerHeight = Handler.getOuterHeight(this.dt.el.nativeElement.children[0]);
 
-        relativeHeight = Handler.getOuterHeight(this.dt.el.nativeElement.parentElement) * parseInt(this.scrollHeight, 10) / 100;
+        if (this.scrollHeight.includes('calc')) {
```

Comment:
Could you use a variable to store this.scrollBodyViewChild and this.scrollHeight to make the line shorter. This can also saving the searching variable time when running code.